### PR TITLE
Protocol Freeze (Beta 100)

### PR DESCRIPTION
**Pull request by pedrouid**
FINAL PROTOCOL CHANGES 😆 

This PR is intended to be the freeze all protocol changes for Beta 100 release and includes the following protocol changes:
* removes updateAccounts()
* renames updateNamespaces() with update()
* renames updateExpiry() with extend()
* refactor namespaces with a map and optional extension

### CODE REVIEW

File: auth/data-structures.md [hunk 1]
```diff
@@ -80,14 +80,21 @@ Session is a topic encrypted by a symmetric key derived using a key agreement es
   "expiry": Int64, // timestamp (seconds)
   "acknowledged": boolean,
   "controller": string,
-  "accounts": [string],
-  "namespaces": [
-    {
-      "chains": [string],
+  "namespaces": {
+    [namespace_name] : {
```
Comment (dukevantreta) [inline auth/data-structures.md]:
```suggestion
    "<namespace_name>" : {
```

File: auth/data-structures.md [hunk 2]
```diff
@@ -113,13 +120,20 @@ Proposal is sent by the proposer client to be approved or rejected by the respon
       "icons": [string]
     }
   },
-  "namespaces": [
-    {
+  "requiredNamespaces": {
+    [namespace_name] : {
```
Comment (dukevantreta) [inline auth/data-structures.md]:
```suggestion
    "<namespace_name>" : {
```

File: auth/data-structures.md [hunk 3]
```diff
@@ -168,13 +182,20 @@ Settelement is sent by the responder after approval and it's broadcasted right a
     }
   },
   "accounts": [string], // CAIP-10 account IDs
-  "namespaces": [
-    {
-      "chains": [string],
+  "namespaces": {
+    [namespace_name] : {
```
Comment (dukevantreta) [inline auth/data-structures.md]:
```suggestion
    "<namespace_name>" : {
```

File: auth/rpc-methods.md [hunk 4]
```diff
@@ -47,13 +47,20 @@ Used to propose a session through topic A. Requires a success response with asso
       "icons": [string]
     }
   },
-  "namespaces": [
-    {
+  "requiredNamespaces": {
+    [namespace_name] : {
```
Comment (dukevantreta) [inline auth/rpc-methods.md]:
```suggestion
     "<namespace_name>" : {
```

File: auth/rpc-methods.md [hunk 5]
```diff
@@ -111,50 +118,54 @@ Used to settle a session over topic B.
       "icons": [string]
     }
   },
-  "accounts": [string], // CAIP-10 account IDs
-  "namespaces": [
-    {
-      "chains": [string],
+  "namespaces": {
+    [namespace_name] : {
```
Comment (dukevantreta) [inline auth/rpc-methods.md]:
```suggestion
     "<namespace_name>" : {
```

File: auth/rpc-methods.md [hunk 6]
```diff
@@ -111,50 +118,54 @@ Used to settle a session over topic B.
...
 
 ```jsonc
-// wc_sessionUpdateNamespaces params
+// wc_sessionUpdate params
 {
-  "namespaces": [
-    {
-      "chains": [string],
+  "namespaces": {
+    [namespace_name] : {
```
Comment (dukevantreta) [inline auth/rpc-methods.md]:
```suggestion
     "<namespace_name>" : {
```

**Comment by dukevantreta** [comment]
Adding some suggestions to avoid misunderstanding the namespace name key with an array

File: auth/client-api.md [hunk 7]
```diff
@@ -16,16 +16,15 @@ abstract class Client {
 
   // for proposer to create a session with or without pairing creation
   public abstract connect(params: {
-    namespaces: Namespace[];
+    requiredNamespaces: ProposalNamespace[];
```
Comment (dukevantreta) [inline auth/client-api.md]: An array of namespaces is passed as param, but it is encoded as a dictionary. Where to get the key (namespace_name) from? Do we just get the chain name from the first value of the `chains` or `account` arrays?

**Comment by dukevantreta** [comment]
And after metamask answers: Do we have to support namespaces without a chain / account for beta100?

File: auth/client-api.md (hunk 7 above)
Comment (pedrouid) [reply auth/client-api.md]: Sorry that was a typo, thanks for catching it 🙌  I've fixed it

**Comment by pedrouid** [comment]
Thanks @MisterVants for reviewing it! I've fixed the client api typo and added all the suggested changes

Regarding the MetaMask suggestion of optional chains I think it doesn't bring any value for our protocol so we are still going to require namespaces, request and emit to always have a chain

File: auth/data-structures.md [hunk 8]
```diff
@@ -80,14 +80,21 @@ Session is a topic encrypted by a symmetric key derived using a key agreement es
...
-      "chains": [string],
+  "namespaces": {
+    "<namespace_name>" : {
+      "accounts": [string],
       "methods": [string],
       "events": [string],
+      "extension": [ // optional
+        {
+          "accounts": [string],
+          "methods": [string], // optional
```
Comment (Talhaali00) [inline auth/data-structures.md]: Does it make sense to make methods and events optional? I don't see the point in having just accounts in the extensions block. Would it make more sense to require methods or events along with accounts?

Comment (pedrouid) [reply auth/data-structures.md]: That's a good point. They can just be empty arrays

**Pull request merged**